### ISSUE TITLE
V3: Some Studio logos are corrupted when written to disk

### ISSUE THREAD

**Issue by plz12345** [Whisparr/Whisparr#176]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

When refreshing a studio in V3, the studio logo is loaded on the fly from StashDB.  However, for some Studios, it fails to save the file to disk properly, and as such, the logo won't load after refreshing the page unless the studio is updated again.

I'm able to repro reliably with the `All Her Luv` and `MissaX` studios, if that helps.  When I went into the `.config/Whisparr/MediaCover/studio` and subsequent subfolders, I could confirm that studios with this issue has corrupted PNG files, whereas studios without this issue had no issues previewing the PNG's.  These are the `clearlogo.png` files in each studio ID subfolder.

I don't know .NET well enough to find where in the code this is breaking, but I'd imagine it's either doe to a failure to download (unlikely, since it can happen reliably) or some kind of failed image conversion.  I think it's the latter, as all the corrupt images are less than 10KB in size, whereas working ones are usually larger.

Example
`https://cdn.stashdb.org/images/ff/21/ff21ae40-5119-4139-aecc-40389b7efee7`
Saves locally to me in Studio ID 40, and this URL
`http://localhost:6969/MediaCover/studio/40/clearlogo.png?lastWrite=638461675106038543`
Fails hard and shows this message when directly loading in the browser
>The image “http://localhost:6969/MediaCover/studio/40/clearlogo.png?lastWrite=638461675106038543” cannot be displayed because it contains errors.

### Expected Behavior

Whisparr saves studio logos locally to the app data folder in a readable format.

### Steps To Reproduce

1. Add `All Her Luv` as a studio
2. When the refresh/sync happens, the logo will load as expected
3. Refresh the page
4. Logo disappears because the local file is corrupt

### Environment

```markdown
- OS: MacOS built from source, as well as Linux-based Docker
- Whisparr: 3.0.0.484
- Docker Install: yes
- Using Reverse Proxy: no
- Browser: firefox and edge
- Database: SQLite
```

### What branch are you running?

Develop

### Trace Logs?

https://privatebin.net/?2191de05ca564b66#9MMLZvm9ArRnyLfjJT8TxvSJdXV33P2gDxBjN8pSKXZ4

**Comment by Tom-Estes** [comment]
Seems to happen when the logo is a svg.

Added to HttpClient.cs DownloadFileAsync
    `if (response.Headers.ContentType != null && response.Headers.ContentType.Contains("image/svg+xml"))
{
    throw new HttpException(request, response, "Site responded with svg content.");
}`

and it throws the exception.

We need a way to convert the svg to a png.

**Comment by plz12345** [comment]
Would something like this work?  I don't know what's involved w/r/t adding a library to the project.  https://github.com/svg-net/SVG

**Comment by Tom-Estes** [comment]
I'm sure it will, not sure on if allowed to add it to the dependency.

**Comment by plz12345** [comment]
Is there a reason Whisparr can't just support the SVG format directly?  StashDB itself seems to.

**Comment by Tom-Estes** [comment]
Only glanced at it and checked the file type. The png extension is hardcoded, but I'll see if there's a way to pass the downloaded file extension to the filename.

**Comment by plz12345** [comment]
Excellent, thanks.  It's an uphill battle to get StsahDB consistency on images since it's crowdsourced, and I think it'd more practical to handle in place vs. trying to include additional modules into the app.  I'm sure that'd cause issues later, somehow.

**Comment by Qstick** [comment]
This is fixed in Lidarr, we should align to that

**Comment by plz12345** [comment]
@Qstick are you sure?  It has the same code setting file extension for `clearlogo` to .png.

https://github.com/Lidarr/Lidarr/blob/develop/src/NzbDrone.Core/MediaCover/MediaCoverService.cs

``` C#
private string GetExtension(MediaCoverTypes coverType, string defaultExtension)
        {
            return coverType switch
            {
                MediaCoverTypes.Clearlogo => ".png",
                _ => defaultExtension
            };
        }
```

**Comment by Qstick** [comment]
It's closer I should say. We should use the default extension first (current image extension), and just default to a hard code if that's not available

**Comment by plz12345** [comment]
That would not fix it.  StashDB doesn't actually use extensions in the URL for the image.

**Comment by plz12345** [comment]
I have this about halfway fixed. Just commenting here so no one else works on it.

**Comment by Tom-Estes** [comment]
Nice! Already have a fix, but reworking so we don't hard code an extension when retrieving the file path. I'll stop and let you do your PR.

**Comment by plz12345** [comment]
We were probably at the same point.  I'm re-writing the `DownloadStudioCover` function to be aware of the content-type header, and using that to determine the file extension used.  Once I have that, the `ConvertToLocalStudioUrls` function just needs to parse files in the folder, vs. blindly assuming coverlogo.png is in that folder.

**Comment by plz12345** [comment]
OK, I have this working locally, but in testing it, I think I uncovered a potential separate bug.  It appears that when you delete the last scene of a studio, it doesn't remove the studio.  Same for performers, so this ends up leaving orphan studios and performers that the UI has no ability to directly remove. 

 If that's the case and this is (at least now) known, I can optimize and ship this PR specific to fixing the Studio logos, but will raise a separate issue related to the orphan management.  I want to try laying this on a clean eros build with a clean DB and app data folder structure, before being comfy submitting this.  I had to add some extra error handling but did try to reuse the project-native classes/functions where possible.

Here's a SFW view showing all the logos now loading locally, regardless of image file type/extension.
<img width="1656" alt="image" src="https://github.com/Whisparr/Whisparr/assets/132735020/b6930ef3-deaf-42af-8758-08093c140f55">
